### DISCUSSION TITLE
Sitemap for payloadCMS?

### DISCUSSION THREAD

**Discussion by janbrb** [payloadcms/payload#4920] [Q&A] [answered]
Did anyone solve this already?

I can't find any information on sitemaps and would like to implement it and a rss feed in payload.

**Comment by zubricks** [comment, marked as answer]
Hey @janbrb what are you using on the frontend? If you're using NextJS, there is a [really handy package](https://www.npmjs.com/package/next-sitemap#generating-dynamicserver-side-sitemaps) that we'd suggest.

**Reply by janbrb** [reply]
Hi @zubricks! Thanks for the reply and the link. I had problems with next-sitemap and using the "website" template at the beginning. Got it working now with the dynamic example for nexjs using the app/ folder.

**Reply by BraianPita** [reply]
For further reference for anyone trying this. I was having issues with my sitemap not showing dynamic results from my query to the pages collection even after using the next-sitemap package when building a docker image. I ended up needing the following line:
```ts
export const dynamic = 'force-dynamic'
```
To force my sitemap not to cache the result from build time (No database when building the docker image). Also, I don't know if this is relevant but the way I did it was by creating a GET api route under `app/sitemap.xml/route.ts`. I am unsure if this works with the default nextjs sitemap (see [here](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap))

**Comment by SalahAdDin** [comment]
Is there any plan to create a plugin for such functionality?
